self-paced-labs/network/frr_bgp/README.md

```diff
@@ -396,4 +412,51 @@ replace:
     103.172.124.0/24;
  }
 }
+```
+
+# Terraform 教學
+
+由於本教學是基於 [Terraform](https://www.terraform.io/) 的，所以會使用 [Terraform](https://www.terraform.io/) 的環境來進行教學。
+
+## 安裝 Terraform
+
+至 Terraform 官網的 [Downloads](https://www.terraform.io/downloads.html) 下載最新版本。  
+不同的版本有不同的下載方式 / 指令。
+
+## How to run Terraform script
+
+一旦安裝好 Terraform 後，我們就可以來進行部署了。
+
+首先執行以下指令：
+
+```shell
+# 先初始化環境
+`terraform init`
+```
+
+接著，我們要設定 OpenStack 的環境資訊，修改並執行以下指令。
+
+```shell
+# 新增 OpenStack 資訊
+export OS_USERNAME="" # 輸入 OpenStack 的帳號
+export OS_TENANT_NAME="" # 輸入 OpenStack 的帳號
+export OS_PASSWORD="" # 輸入 OpenStack 的密碼
+export OS_AUTH_URL="https://openstack.cloudnative.tw:5000" # 輸入 OpenStack 的授權網址，可以在 API Access 中找到
```

Comment:
We can download the credential file directly from dashboard, which will be easier in most cases.